packages/babel-plugin-syntax-decorators/src/index.ts

```diff
@@ -2,39 +2,59 @@ import { declare } from "@babel/helper-plugin-utils";
 
 export interface Options {
   legacy?: boolean;
+  // TODO(Babel 8): Remove "2018-09"
   version?: "legacy" | "2018-09" | "2021-12";
   decoratorsBeforeExport?: boolean;
 }
 
+const VALID_VERSIONS = process.env.BABEL_8_BREAKING
+  ? new Set(["2021-12", "legacy"])
+  : new Set(["2021-12", "2018-09", "legacy"]);
+
 export default declare((api, options: Options) => {
   api.assertVersion(7);
 
-  const {
-    legacy, // TODO: Remove in Babel 8
+  if (!process.env.BABEL_8_BREAKING) {
+    const { legacy } = options;
+    // eslint-disable-next-line no-var
+    var { version = legacy ? "legacy" : "2018-09" } = options;
 
-    version = legacy ? "legacy" : "2018-09",
-    decoratorsBeforeExport = version === "2021-12" ? false : undefined,
-  } = options;
-  if (version !== "2021-12" && version !== "2018-09" && version !== "legacy") {
-    throw new Error("Unsupported decorators version: " + version);
-  }
-  if (legacy !== undefined) {
-    if (typeof legacy !== "boolean") {
-      throw new Error(".legacy must be a boolean.");
-    }
-    if (options.version !== undefined) {
-      throw new Error(
-        "You can either use the .legacy or the .version option, not both.",
-      );
+    if (legacy !== undefined) {
+      if (typeof legacy !== "boolean") {
+        throw new Error(".legacy must be a boolean.");
+      }
+      if (options.version !== undefined) {
+        throw new Error(
+          "You can either use the .legacy or the .version option, not both.",
+        );
+      }
     }
+  } else {
+    // eslint-disable-next-line no-var
+    var { version } = options;
+  }
+  if (version === undefined) {
+    throw new Error(
+      "The decorators plugin requires a 'version' option, whose value must be one of: " +
+        Array.from(VALID_VERSIONS, v => `'${v}'`).join(", "),
+    );
+  }
+  if (!VALID_VERSIONS.has(version)) {
+    throw new Error("Unsupported decorators version: " + version);
   }
 
+  let { decoratorsBeforeExport } = options;
   if (decoratorsBeforeExport === undefined) {
-    if (version === "2018-09") {
-      throw new Error(
-        "The decorators plugin, when .version is '2018-09' or not specified," +
-          " requires a 'decoratorsBeforeExport' option, whose value must be a boolean.",
-      );
+    if (version === "2021-12") {
+      decoratorsBeforeExport = false;
```

Comment:
Does it mean we can remove the `decoratorsBeforeExport` generator option in Babel 8?

Reply:
This PR still keeps the `decoratorsBeforeExport` option but makes it default to `false`, I think I can just remove it given the proposal is now stage 3.